lib/ansible/runner/response.rb

```diff
@@ -79,9 +80,25 @@ def load_return_code
         1
       end
 
+      # Match a full json object that starts "somewhere" on the line, and ends
+      # at the end.
+      RUNNER_STDOUT_LINE_MATCH = /^[^\{]*(\{.*\})$/
+
       # @return [String] Stdout that is text, where each line should be JSON encoded object
       def load_stdout
-        File.read(File.join(base_dir, "artifacts", ident, "stdout"))
+        "".tap do |stdout|
+          # Read the stdout file line by line, and filter out lines that don't
+          # include a full JSON object that ends at the end of the line.
+          #
+          # FIXME:  This isn't terribly fool proof, and most likely error
+          # prone.
+          File.foreach(File.join(base_dir, "artifacts", ident, "stdout")) do |line|
```

Comment:
Does this read line-by-line without loading the whole file in memory?

Reply:
https://ruby-doc.org/core-2.4.1/IO.html#method-c-foreach

> Executes the block for every line in the named I/O port, where lines are separated by sep.